app/controllers/vm_common.rb

```diff
@@ -115,14 +115,22 @@ def launch_vmware_console
                   :mks_classid => get_vmdb_config[:server][:mks_classid]
                 )
               when "vmrc"
+                host = @record.ext_management_system.ipaddress || @record.ext_management_system.hostname
+                ticket = j(params[:ticket])
+                vmid = @record.ems_ref
                 {
-                  :host        => @record.ext_management_system.ipaddress ||
-                                  @record.ext_management_system.hostname,
+                  :host        => host,
                   :vmid        => @record.ems_ref,
-                  :ticket      => j(params[:ticket]),
+                  :ticket      => ticket,
                   :api_version => @record.ext_management_system.api_version.to_s,
                   :os          => browser_info(:os),
-                  :name        => @record.name
+                  :name        => @record.name,
+                  :vmrc_uri    => URI::Generic.build(:scheme   => "vmrc",
+                                                     :userinfo => "clone:#{ticket}",
+                                                     :host     => host,
+                                                     :port     => 443,
+                                                     :path     => "/",
+                                                     :query    => "moid=#{vmid}")
```

Comment:
Does this work correctly or do you have to call .to_s manually (I'm assuming it works if you tested it :) )